incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/streams/OpenapiServerFactory.java

```diff
@@ -0,0 +1,1066 @@
+    private final LongUnaryOperator supplyReplyId;
+    private final LongSupplier supplyTraceId;
+    private final Long2ObjectHashMap<OpenapiBindingConfig> bindings;
+    private final int openapiTypeId;
+    private final int httpTypeId;
+
+
+    public OpenapiServerFactory(
+        OpenapiConfiguration config,
+        EngineContext context)
+    {
+        this.writeBuffer = context.writeBuffer();
+        this.extBuffer = new UnsafeBuffer(new byte[writeBuffer.capacity()]);
+        this.bufferPool = context.bufferPool();
+        this.streamFactory = context.streamFactory();
+        this.supplyInitialId = context::supplyInitialId;
+        this.supplyReplyId = context::supplyReplyId;
+        this.supplyTraceId = context::supplyTraceId;
+        this.bindings = new Long2ObjectHashMap<>();
+        this.openapiTypeId = context.supplyTypeId(OpenapiBinding.NAME);
+        this.httpTypeId = context.supplyTypeId(HTTP_TYPE_NAME);
+    }
+
+    @Override
+    public int originTypeId()
+    {
+        return httpTypeId;
+    }
+
+    @Override
+    public int routedTypeId()
+    {
+        return openapiTypeId;
+    }
+
+    @Override
+    public void attach(
+        BindingConfig binding)
+    {
+        OpenapiBindingConfig openapiBinding = new OpenapiBindingConfig(binding);
+        bindings.put(binding.id, openapiBinding);
+    }
+
+    @Override
+    public void detach(
+        long bindingId)
+    {
+        bindings.remove(bindingId);
+    }
+
+    @Override
+    public MessageConsumer newStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length,
+        MessageConsumer receiver)
+    {
+        final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+        final long originId = begin.originId();
+        final long routedId = begin.routedId();
+        final long initialId = begin.streamId();
+        final long affinity = begin.affinity();
+        final long authorization = begin.authorization();
+        final OctetsFW extension = begin.extension();
+        final HttpBeginExFW httpBeginEx = extension.get(httpBeginExRO::tryWrap);
+
+        final OpenapiBindingConfig binding = bindings.get(routedId);
+
+        MessageConsumer newStream = null;
+
+        if (binding != null && binding.isCompositeBinding(originId))
+        {
+            final String operationId = binding.resolveOperationId(httpBeginEx);
+
+            OpenapiRouteConfig route = binding.resolve(authorization);
+
+            if (route != null)
+            {
+                newStream = new HttpStream(
+                    receiver,
+                    originId,
+                    routedId,
+                    initialId,
+                    affinity,
+                    authorization,
+                    route.id,
+                    operationId)::onHttpMessage;
+            }
+
+        }
+
+        return newStream;
+    }
+
+    private MessageConsumer newStream(
```

Comment:
Let's move these pseudo-static `newStream` and `doXxx` helper methods to the end of the factory for improved readability.